### PR TITLE
Commander: bugfix: get all available messages on the subsystem_info t…

### PR DESCRIPTION
**Pull request by dusan19**
…opic

Bugfix when getting all available messages on the subsystem_info topic in commander

### CODE REVIEW

File: src/modules/commander/Commander.cpp [hunk 1]
```diff
@@ -1737,7 +1737,7 @@ Commander::run()
 
 		/* update subsystem info which arrives from outside of commander*/
 		do {
-			if (subsys_sub.updated()) {
+			if ( (updated=subsys_sub.updated()) ) {
```
Comment (julianoes) [inline src/modules/commander/Commander.cpp]: Wow,  this `updated` bool is defined 500 lines above. It looks like it's not used anywhere else. Could you move it down to here where it's used?

Comment (julianoes) [reply src/modules/commander/Commander.cpp]: Actually, can't you just simplify it like this?

```
		/* update subsystem info which arrives from outside of commander*/
		while (subsys_sub.updated()) {
			subsystem_info_s info{};
			subsys_sub.copy(&info);
			set_health_flags(info.subsystem_type, info.present, info.enabled, info.ok, status);
			status_changed = true;
		}
```

Comment (dusan19) [reply src/modules/commander/Commander.cpp]: yes, changed

Comment (dusan19) [reply src/modules/commander/Commander.cpp]: also tested, it gets at most 5 messages in 1 commander loop, which is limited by the messages orb queue size

File: src/modules/commander/Commander.cpp [hunk 2]
```diff
@@ -1736,14 +1734,12 @@ Commander::run()
...
 		/* update subsystem info which arrives from outside of commander*/
-		do {
-			if (subsys_sub.updated()) {
-				subsystem_info_s info{};
-				subsys_sub.copy(&info);
-				set_health_flags(info.subsystem_type, info.present, info.enabled, info.ok, status);
-				status_changed = true;
-			}
-		} while (updated);
+		while (subsys_sub.updated())  {
```
Comment (bkueng) [inline src/modules/commander/Commander.cpp]: You can further simplify to:
```suggestion
		subsystem_info_s info;
		while (subsys_sub.update(&info))  {
```

File: src/modules/commander/Commander.cpp [hunk 3]
```diff
@@ -1736,14 +1734,11 @@ Commander::run()
...
 		/* update subsystem info which arrives from outside of commander*/
-		do {
-			if (subsys_sub.updated()) {
-				subsystem_info_s info{};
-				subsys_sub.copy(&info);
-				set_health_flags(info.subsystem_type, info.present, info.enabled, info.ok, status);
-				status_changed = true;
-			}
-		} while (updated);
+		subsystem_info_s info{};
```
Comment (bkueng) [inline src/modules/commander/Commander.cpp]: No need to initialize:
```suggestion
		subsystem_info_s info;
```

**Pull request merged**